### ISSUE TITLE
None-value problems that arose

### ISSUE THREAD

**Issue by John-Athanasopoulos** [DataStories-UniPi/miniDB#160]
Some problems arose with the use of None-values when rows were deleted from the database tables. Specifically, the functionalities that were affected are:

1. Incorrect inner and SM joins. (These ones have been corrected by me as they were my responsibility. If needed, I can revert the changes).
2. Incorrect INL join (probably because of **3**)
3. Cannot create index on tables that contain None values.
4. When we use the top k parameter in **select** if a value of the first k is None only k-1 values will be presented, if 2 values are None k-2 will be presented etc.
5. Probably the order by functionality will be wrong (it is not working for me at the moment, even for tables with no None values, but I didn't try to correct it).

I think that's all, but other functionalities might get affected in a form of a chain reaction.

**Comment by giorgostheo** [comment]
Hey,

No. 3 can be easily solved by adding this to line 666. I missed it somehow. That should take care of 1-3.
`if key is None: continue`

No. 4 is true and solving is without using a "hacky" way is not trivial. I will look into it.

No. 5 is not true for the master version. Maybe you should try to find out what was added that caused this bug

**Comment by John-Athanasopoulos** [comment]
Hello,
That's correct, the **3rd** issue was resolved easily with the addition you suggested. However the **second** issue (INLJ) still fails, so some debugging and fixing is needed (I will have it completed in a few days time). It probably happens because it tries to compare the None values of a table with the values of the index.

Apart from that, I will try to troubleshoot and find the reason behind the 5th issue's failure.

To summarize: 1 and 3 are completed, I will look into 2 and 5 and, finally, I will wait for you to decide how the 4th issue should be approached.

I think that's all for now :)
